tests/partitions_test.py

```diff
@@ -227,34 +238,31 @@ def test_learner_does_not_take_part(network, args):
                 timeout=ceil(args.join_timer * 2 / 1000),
                 valid_from=str(infra.crypto.datetime_to_X509time(datetime.now())),
             )
-            new_node.wait_for_node_to_join(timeout=ceil(args.join_timer * 2 / 1000))
-            join_failed = False
-        except Exception:
-            join_failed = True
-
-        if not join_failed:
-            raise Exception("join succeeded unexpectedly")
-
-        with new_node.client(self_signed_ok=True) as c:
-            r = c.get("/node/network/nodes/self")
-            assert r.body.json()["status"] == "Learner"
-            r = c.get("/node/consensus")
-            assert new_node.node_id in r.body.json()["details"]["learners"]
-
-        # New node joins, but cannot be promoted to TRUSTED without f other backups
+        except TimeoutError:
+            LOG.info("Trust node proposal did not commit as expected")
+        else:
+            raise Exception("Trust node proposal committed unexpectedly")
 
         check_does_not_progress(primary, timeout=5)
 
+        LOG.info("New joiner is not promoted to Trusted without f other backups")
         with new_node.client(self_signed_ok=True) as c:
             r = c.get("/node/network/nodes/self")
             assert r.body.json()["status"] == "Learner"
             r = c.get("/node/consensus")
             assert new_node.node_id in r.body.json()["details"]["learners"]
 
-    network.wait_for_primary_unanimity()
-    primary, _ = network.find_nodes()
+    LOG.info("Partition is lifted, wait for primary unaminity on original nodes")
+    # Note: Because trusting the new node failed, the new node is not considered
+    # in the primary unanimity. Indeed, its transition to Trusted may have been rolled back.
+    primary = network.wait_for_primary_unanimity()
     network.wait_for_all_nodes_to_commit(primary=primary)
+
+    LOG.info("Trust new joiner again")
+    network.trust_node(new_node, args)
```

Comment:
Just a thought: The fact that the majority partition makes commit progress is not completely set in stone (as discussed, only happens currently because of the timeout signature when first elected). If the partition time ever drops far enough, or we fix the timeout signature behaviour, then when the partition drops it is possible for 0 to win a future election and commit the original trust proposal, in which case this call will presumably fail? Probably worth adding an explicit `check_can_progress(f_backups[0])` while the partition is up.

Reply:
Good point. We should wait for a new partition primary to be elected before that too. I think that the `transition_node_to_trusted` proposal should be idempotent anyway, but it's better/clearer to add this here too.

Reply:
I think the proposal is idempotent as it checks `if (nodeInfo.status === "Pending") ... ` is there another check somewhere?

Reply:
@wintersteiger Good spot. It definitely is idempotent then.